### ISSUE TITLE
Uncaught TypeError: this.children[k].updateElement is not a function

### ISSUE THREAD

**Issue by ptyskju** [jsonform/jsonform#315]
Hi!
There is an error in for in loop near 2715 line:
```js
  for(var k in this.children) {
    this.children[k].updateElement(this.el || domNode);
  }
```

When iterating through JS object, `var k` at some point is a `this.children` method like `max`, `min` etc.  That loop should look like this:

```js
for (const k in  this.children) {
    if (this.children.hasOwnProperty(k) == false) {
        continue;
    }
    this.children[k].updateElement(this.el || domNode);
}

**Comment by tchapi** [comment]
Hi @ptyskju and thanks for this report.

Do you have an example schema where this problem occurs ?

Thanks

**Comment by ptyskju** [comment]
Sure, here:

```json
{
   "schema":{
      "Id":{
         "type":"string",
         "title":null
      },
      "Login":{
         "type":"string",
         "title":"Login"
      },
      "Password":{
         "type":"string",
         "title":"Password"
      },
      "TargetSite":{
         "type":"string",
         "title":"External URL"
      },
      "AdditionalInfo":{
         "type":"string",
         "title":"More info"
      }
   },
   "form":[
      {
         "key":"Id",
         "type":"hidden"
      },
      {
         "key":"Login",
         "type":"text"
      },
      {
         "key":"Password",
         "type":"text"
      },
      {
         "key":"TargetSite",
         "type":"text"
      },
      {
         "key":"AdditionalInfo",
         "type":"textarea"
      }
   ],
   "value":{
      "Id":null,
      "Login":null,
      "Password":null,
      "TargetSite":null,
      "AdditionalInfo":null
   }
}
```

**Comment by tchapi** [comment]
I have tested your schema in the playground and it works quite well on my side — or maybe I don't see the problem. The values get extracted and there is no error in the console. Am I missing something ?

**Comment by ptyskju** [comment]
I'm using your library via yarn - maybe that cause problems? Detailed error:

```
Uncaught TypeError: this.children[k].updateElement is not a function
    at formNode.push../node_modules/jsonform/lib/jsonform.js.formNode.updateElement (jsonform.js:2715)
    at formNode.push../node_modules/jsonform/lib/jsonform.js.formNode.updateElement (jsonform.js:2715)
    at formNode.push../node_modules/jsonform/lib/jsonform.js.formNode.setContent (jsonform.js:2677)
    at formNode.push../node_modules/jsonform/lib/jsonform.js.formNode.render (jsonform.js:2630)
    at formTree.push../node_modules/jsonform/lib/jsonform.js.formTree.render (jsonform.js:3461)
    at jQuery.fn.init.push../node_modules/jsonform/lib/jsonform.js.$.fn.jsonForm (jsonform.js:3717)
    at Object.<anonymous> (loadFormPart.js:15)
    at fire (jquery.js:3496)
    at Object.fireWith [as resolveWith] (jquery.js:3626)
    at done (jquery.js:9786)
```

Source file:

```js
import jQuery from 'jquery';
import "jsonform";


(function ($) {
    $.fn.loadFormPart = function (url, ajaxData = null, method = "GET") {
        const elementToAddForm = $(this);

        return $.ajax({
            url: url,
            data: ajaxData,
            method: method,
        }).done(function (jsonForm) {
            elementToAddForm.trigger('beforeload.jsonForm');
            elementToAddForm.jsonForm(jsonForm);
            elementToAddForm.trigger('afterload.jsonForm');
        });
    }
})(jQuery);
```

Form by itself loads correctly (see below). Because form rendering is in _done_ function in ajax, error from for-in loop cause endless ajax spinner.  
![image](https://user-images.githubusercontent.com/24554387/105973115-a41d3e80-608c-11eb-939e-72cc45b35202.png)

**Comment by mrszmoline** [comment]
Hello - I am also experiencing the same error message and using the library via yarn. Any fix for this?

**Comment by tchapi** [comment]
I don't see how yarn would affect the functionality of a loop like this one — `this.children` looks like it is a different object with methods when you use your code. I'd like to see a complete reproducible example if possible ?

**Comment by mrszmoline** [comment]
@tchapi  Here is an example in my code. Not sure how it compares to @ptyskju, but the solution of: 
```
 for (const k in  this.children) {
    if (this.children.hasOwnProperty(k) == false) {
        continue;
    }
    this.children[k].updateElement(this.el || domNode);
} 
```
 provided by @ptyskju fixed my problem. 

-- note: I am using ember.js as a framework (I don't think that should matter)

Here is my js
```
import $ from 'jquery';
    import 'jsonform';
    $('form').jsonForm(
      {
        "schema": {
          "name": {
            "title": "Name",
            "description": "Nickname allowed",
            "type": "string"
          },
          "gender": {
            "title": "Gender",
            "description": "Your gender",
            "type": "string",
            "enum": [
              "male",
              "female",
              "alien"
            ]
          }
        }
      }
    ) 
```
Error message in the console: 
![image](https://user-images.githubusercontent.com/35612979/108761425-75bc4300-7514-11eb-89c4-f029a7e9ca21.png)

Taking a look further and doing a 
```
console.log(this.children[k]) 
```
on line 2721
 Here is what I see now in the console, in addition to the error message
![image](https://user-images.githubusercontent.com/35612979/108761846-fbd88980-7514-11eb-8dcd-00f1def35688.png)

Seems it is looping through a second "empty" function? 

Like I mentioned before @ptyskju suggestion fixed my issue. If I need to do a pull request with his suggestion please let me know. 

Thank you
Afton

**Comment by tchapi** [comment]
Closed by #326
